FitGoal/FitGoal/Singup&Singin/SocialMediaAuthenticator.swift

```diff
@@ -53,4 +69,60 @@ class SocialMediaAuthenticator: NSObject, GIDSignInDelegate {
             completion?(.success(()))
         }
     }
+    
+    //MARK: - Facebook
+    
+    private func facebookSignIn(sender: UIViewController, completion: @escaping SignInCallback) {
+        let manager = LoginManager()
+        manager.logIn(permissions: ["public_profile", "email"], from: sender) { (result, error) in
+            if let error = error {
+                completion(.failure(error))
+            } else {
+                guard let result = result else { return }
+                if result.isCancelled {
+                    let error = LoginError.userCanceledLogIn
+                    completion(.failure(error))
+                }
+                guard let accessToken = AccessToken.current else { return }
+                let credentials = FacebookAuthProvider.credential(withAccessToken: accessToken.tokenString)
+                
+                /*This part of the function looks for a previous user logged in with these credentials but with a different providers.
+                 Because of the behavior of the GoogleSignIn, it looks as if this functionality is implicit at some point
+                */
+                if let user = Auth.auth().currentUser {
+                    guard let name = user.displayName else { return }
+                    guard let email = user.email else { return }
+                    let userInfo = UserInformation(name: name, email: email)
+                    self.appPreferences.loggedInUser = userInfo
+                    Auth.auth().fetchSignInMethods(forEmail: email, completion: nil)
+                    user.link(with: credentials, completion: nil)
+                    completion(.success(()))
+                } else {
+                    Auth.auth().signIn(with: credentials) { (user, error) in
+                        if let error = error {
+                            completion(.failure(error))
+                        } else {
+                            guard let user = Auth.auth().currentUser else { return }
+                            guard let name = user.displayName else { return }
+                            guard let email = user.email else { return }
+                            let userInfo = UserInformation(name: name, email: email)
+                            self.appPreferences.loggedInUser = userInfo
+                            completion(.success(()))
```

Comment:
This block of code is the same as in the `if` branch above. Could you please move it to a function so that the logic is written only once?